lib/update/phase_app.go

```diff
@@ -190,11 +192,25 @@ func (p *phaseApp) runHooks(ctx context.Context, hooks ...schema.HookType) error
 			}
 			return trace.Wrap(err)
 		}
-		_, out, err := app.RunAppHook(ctx, p.Apps, req)
+		reader, writer := io.Pipe()
+		go func() {
+			defer reader.Close()
+			scanner := bufio.NewScanner(reader)
+			for scanner.Scan() {
+				log.WithField(trace.Component, "hook").Info(scanner.Text())
```

Comment:
I think the streaming part should be factored out and use the name of the hook in the log at the very least:
```go
func (p *phaseApp) runHooks(ctx context.Context, hooks ...schema.HookType) error {
	// ...
	reader, writer := io.Pipe()
	go streamHook(hook, reader)
	// ...
}

func streamHook(hook string, rd io.ReadCloser) error {
	defer rc.Close()
	scanner := bufio.NewScanner(rc)
	log := log.WithFields(log.Fields{trace.Component: "hook", "hook": hook})
	for scanner.Scan() {
		log.Info(scanner.Text())
	}
	if err := scanner.Err(); err != nil {
		return trace.Wrap(err)
	}
	return nil
}
```

Reply:
I actually did that initially (added "hook" field) but it was always hook=update so I doubted its usefulness and removed. I can bring it back though.